webapp/src/components/modals/channel_subscriptions/edit_channel_subscription.tsx

```diff
@@ -275,9 +340,14 @@ export default class EditChannelSubscription extends PureComponent<Props, State>
             this.fetchIssueMetadata(projects, this.state.instanceID);
         }
 
+        if (projects && projects.length) {
+            this.fetchIssueStatuses(projects[0], this.state.instanceID);
```

Comment:
We can put this in the same if block as above

Also, we could include the statuses in the response to `fetchIssueMetadata` instead of creating a new endpoint right? The statuses are currently only fetched at the same spot as that call anyway. That may simplify code a bit and require less code for this feature

Reply:
Another approach would be to encapsulate the fetching and state management to a separate component. This way the parent component doesn't need to change at all. This is currently how the epic link selector works by delegating the state management to the `BackendSelector` component.

https://github.com/mattermost/mattermost-plugin-jira/blob/d1e233af901f5a4c3b652fee260857c74d1c6d44/webapp/src/components/data_selectors/jira_epic_selector/jira_epic_selector.tsx#L100-L106